encoding/dabatch.go

```diff
@@ -0,0 +1,244 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+)
+
+// DABatchBase contains common metadata for all versions of DABatch
+type DABatchBase struct {
+	Version                uint8
+	BatchIndex             uint64
+	L1MessagePopped        uint64
+	TotalL1MessagePopped   uint64
+	DataHash               common.Hash
+	ParentBatchHash        common.Hash
+	SkippedL1MessageBitmap []byte
+}
```

Comment:
all fields of any type that we're expecting to handle solely via interface should be private and exposed via interface, otherwise they're not accessible without casting into the respective type

Reply:
will resolve this general problem in other comments, thus resolving this comment first. since `DABatchBase` is removed.